src/Reflex/Class.hs

```diff
@@ -298,6 +302,18 @@ class ( MonadHold t (PushM t)
   mergeIntIncremental :: Incremental t (PatchIntMap (Event t a)) -> Event t (IntMap a)
   fanInt :: Event t (IntMap a) -> EventSelectorInt t a
 
+-- | Efficiently fan-out an event to many destinations. You should save the
+-- result in a @let@-binding, and then repeatedly 'select' on the result to
+-- create child events
+fan :: forall t k. (Reflex t, GCompare k)
+    => Event t (DMap k Identity) -> EventSelector t k
+    --TODO: Can we help enforce the partial application discipline here?  The combinator is worthless without it
+fan e = EventSelector (fixup (selectg (fanG e) :: k a -> Event t (Identity a)) :: forall a. k a -> Event t a)
```

Comment:
Should we match on the `Coercion` and then produce a lambda, or something like that, to make sure there's only one `eventCoercion` call per `Reflex` instance?

Reply:
It probably doesn't actually matter...

Reply:
I was thinking about such things and opened https://github.com/reflex-frp/reflex/issues/319